### DISCUSSION TITLE
Surface super-slow when battery is empty

### DISCUSSION THREAD

**Discussion by Freek07** [linux-surface/linux-surface#1145] [General]
I assume it does not relate directly to the project, but it's a surface issue nevertheless, so you might be able to help.

My SP7 works great with Linux, but when I boot it  (connected to a charger!) with next-to-empty battery, it's really sluggish. I tried tweaking TLP, installed cpufreq-daemon, but it's not the CPU, as it boosts (just checked, and one of the cores actually boosted to 1,5GHz now (i5-1035)), but the system is still ultra-slow and non-responsive.

After a reboot (when the battery is more charged), it will (re)act normally and responsively. 
I suspect it must be some power-saving stuff.

Does anyone have an idea how to make it work "normally" without rebooting?

Thanks.

**Comment by hydrandt** [comment]
Same here. Surface Go (first generation), Ubuntu 23.10 with the linux-surface kernel. If the battery got to low charge percentage (I don't know the exact number here, but I think it is around 20%), the system becomes super sluggish (for example, switching between gnome desktops becomes very choppy, when normally it is pretty smooth), and stays sluggish even when it is charged again.

Would be happy for any hints - what to look at.

I only tried to change the performance profiles as it is possible in the top right menu in Gnome.

**Reply by StollD** [reply]
I never really investigated this, but I think by SB2 gets sluggish when the battery goes below 30%, no matter if it is being charged. So it stops being sluggish when you charge it back to more than 30%.

I also vaguely remember that suspending and then resuming can fix it, but I don't know how accurate that is.

**Reply by hydrandt** [reply]
Thanks for the feedback. I'll observe:)
